src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -222,13 +211,92 @@ public sealed class ReadyToRunCodegenNodeFactory : NodeFactory
                   nameMangler,
                   new ReadyToRunTableManager(context))
         {
-            _importMethods = new Dictionary<TypeAndMethod, IMethodNode>();
 
             Resolver = moduleTokenResolver;
             InputModuleContext = signatureContext;
             CopiedCorHeaderNode = corHeaderNode;
             if (!win32Resources.IsEmpty)
                 Win32ResourcesNode = new Win32ResourcesNode(win32Resources);
+
+            // Create node caches
```

Comment:
>// Create node caches [](start = 12, length = 21)

It would be better to bring back the CreateNodeCaches method, and move all of these creations to it. This will also match the design we had for .Net Native

Reply:
That prevents us from making the NodeCaches readonly, since they're not being assigned in a constructor anymore.

Reply:
I think it's fine to not make them readonly

Reply:
It's fine not to make them readonly. At the end of the day, I think we should be consistent with how we implement these, so given that we have other places that initialize the NodeCaches inside a CreateNodeCaches function, we should do the same here